presto-hive/src/main/java/com/facebook/presto/hive/authentication/KerberosHadoopAuthentication.java

```diff
@@ -25,24 +26,36 @@
 public class KerberosHadoopAuthentication
         implements HadoopAuthentication
 {
-    static {
+    private final KerberosAuthentication kerberosAuthentication;
+
+    public static KerberosHadoopAuthentication createKerberosHadoopAuthentication(KerberosAuthentication kerberosAuthentication, HdfsConfiguration hdfsConfiguration)
+    {
+        KerberosHadoopAuthentication kerberosHadoopAuthentication = new KerberosHadoopAuthentication(kerberosAuthentication);
+
+        Configuration configuration = requireNonNull(hdfsConfiguration, "hdfsConfiguration is null").getConfiguration();
+        initializeConfigs(configuration);
+        return kerberosHadoopAuthentication;
```

Comment:
How `kerberosHadoopAuthentication` is bound with `hdfsConfiguration`? They are not used in the same context within this method. Is this via some static variables?

Reply:
Unfortunately yes.